### PR TITLE
Add collision support and refactoring coct member/class names

### PR DESCRIPTION
**Pull request by kenjiuno**
WIP

Adding bar type `6 Map Collision Data` support to OpenKh.

`OpenKh.Kh2.Coct`: Low level reader/writer of COCT
`OpenKh.Kh2.CoctLogical`: A high standard builder of coct

`OpenKh.Tools.Kh2MapCollisionEditor` developed by Xeeynamo.

2020-07-20_21h18_24

`OpenKh.Command.CoctChanger` tool

```bat
OpenKh.Command.CoctChanger.exe

1.0.0

Usage: OpenKh.Command.CoctChanger [command] [options]

Options:
  --version         Show version information
  -?|-h|--help      Show help information

Commands:
  create-room-coct  coct file: create single closed room
  use-this-coct     map file: replace coct with your coct

Run 'OpenKh.Command.CoctChanger [command] -?|-h|--help' for more information about a command.
```

`create-room-coct` command

```bat
OpenKh.Command.CoctChanger.exe create-room-coct -h

coct file: create single closed room

Usage: OpenKh.Command.CoctChanger create-room-coct [options] <CoctOut>

Arguments:
  CoctOut           Output coct

Options:
  -?|-h|--help      Show help information
  -b|--bbox <BBOX>  bbox in model 3D space: minX,Y,Z,maxX,Y,Z (default: ...)
```

```bat
OpenKh.Command.CoctChanger.exe create-room-coct room.coct --bbox -1800,-500,-1800,1800,1000,1800

```

`use-this-coct` command

```bat
OpenKh.Command.CoctChanger.exe use-this-coct C:\A\eh18 H:\Proj\pcsx2\bin\inject.f266b00b\map\jp room.coct
C:\A\eh18\eh18.map

```

test play on pcsx2:

image

### CODE REVIEW

**Comment by Xeeynamo** [comment]
Hi :) thank you so much to have resumed that branch. I was thinking to change the front-end for the collision editing: I found WPF inappropriate as I would like to preview the map with the collision itself (similar to the approach you used for your toolkit). Do you mind if I occasionally push some changes here?

**Comment by kenjiuno** [comment]
Hi, xeeynamo! thanks for seeing this.
Yes, please push your new changes to this branch `feature/kh2-collisions-plus` directly.

File: docs/tool/index.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 |[2LD](../kh2/file/type/2ld.md#layout) | [OpenKh.Tools.LayoutEditor](./LayoutEditor/index.md)
 |[2DD](../kh2/file/type/2ld.md#sequence) | [OpenKh.Tools.LayoutEditor](./LayoutEditor/index.md)
 |[DOCT](../kh2/file/type/doct) | [OpenKh.Command.DoctChanger](./DoctChanger/index.md)
+|[COCT](../kh2/file/type/coct) | [OpenKh.Command.CoctChanger](./CoctChanger/index.md)
```
Comment (Delta-473) [inline docs/tool/index.md:24]: Can you add .md to the DOCT and COCT link?

Comment (kenjiuno) [reply docs/tool/index.md]: Thanks I'll fix it.

**Pull request merged**